### PR TITLE
Upgrade gradle wrapper to use version 4.0.2

### PR DESCRIPTION
**Pull request by HappyRay**
See
https://docs.gradle.org/current/release-notes.html

Updated using the command:

```
./gradlew wrapper
```

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#Sun May 07 08:40:37 PDT 2017
+#Sat Jun 17 07:21:26 PDT 2017
```
Comment (chengren311) [inline gradle/wrapper/gradle-wrapper.properties]: why do we need this? can we just remove this line?

Comment (HappyRay) [inline gradle/wrapper/gradle-wrapper.properties]: It's autogenerated. And it documents when the file was generated.
I don't think we should manually edit this file.

**Comment by HappyRay** [comment]
Thanks. I will wait a bit until I hear back from the tools team on the potential impact for integration with internal toolings.

**Comment by suvodeep-pyne** [comment]
@HappyRay should we merge this now?

**Comment by suvodeep-pyne** [comment]
It seems `4.0.2` is the most stable in the `4.0.x` pipeline
https://docs.gradle.org/4.0.2/release-notes.html

No known issues. List of Releases: https://services.gradle.org/distributions/

File: build.gradle [hunk 2]
```diff
@@ -171,11 +171,27 @@ subprojects {
     }
 }
 
+/*
```
Comment (suvodeep-pyne) [inline build.gradle]: This should probably be inserted at the top if at all? Do we have to include copyright notices for build scripts?

**Pull request merged**